File: json/json-annotation-processor/src/main/java/ru/tinkoff/kora/json/annotation/processor/reader/JsonClassReaderMeta.java
package ru.tinkoff.kora.json.annotation.processor.reader;

import com.squareup.javapoet.TypeName;

import javax.annotation.Nullable;
import javax.lang.model.element.TypeElement;
import javax.lang.model.element.VariableElement;
import javax.lang.model.type.TypeMirror;
import java.util.List;

public record JsonClassReaderMeta(TypeMirror typeMirror, TypeElement typeElement, List<FieldMeta> fields) {
    public record FieldMeta(VariableElement parameter, String jsonName, TypeName typeName, @Nullable ReaderFieldType typeMeta, @Nullable TypeMirror reader) {}
}

File: json/json-symbol-processor/src/main/kotlin/ru/tinkoff/kora/json/ksp/reader/JsonClassReaderMeta.kt
package ru.tinkoff.kora.json.ksp.reader

import com.google.devtools.ksp.symbol.KSClassDeclaration
import com.google.devtools.ksp.symbol.KSType
import com.google.devtools.ksp.symbol.KSValueParameter
import com.squareup.kotlinpoet.TypeName

data class JsonClassReaderMeta(val classDeclaration: KSClassDeclaration, val fields: List<FieldMeta>) {

    data class FieldMeta(val parameter: KSValueParameter, val jsonName: String, val type: TypeName, val typeMeta: ReaderFieldType, val reader: KSType?)
}
